Convert TZ target name 'NPSA' to test spec platform name

ccli8 opened the pull request:
### Description

This PR is continuation of #11467 and to convert TZ target name with suffix **NPSA** https://github.com/ARMmbed/mbed-os/pull/11288#discussion_r324173785 to test spec platform name. With this PR, TZ targets are named as follows:

1.  All TZ targets should have name pattern: **PLATFORM_[PSA_/NPSA_]S/NS**, where:
    1.  **PLATFORM** for test spec platform name
    1.  **PSA/NPSA** for PSA/non-PSA targets. Defaults to PSA target on absent.
    1.  **S/NS** for secure/non-secure targets
1. Secure target may participate in Greentea, so its name is also truncated here.

#### Related PRs

Continuation of #11467 
Required by https://github.com/ARMmbed/mbed-os/pull/11288#discussion_r324173785

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@devran01 

### Release Notes

Define naming rule for TrustZone targets: 'PLATFORM'\_['NPSA\_']'S'/'NS', where:
1. 'PLATFORM' for test spec platform name, which is registered in platform database of mbed-os-tools.
2. 'NPSA' for non-PSA targets. Defaults to PSA target if absent.
3. 'S'/'NS' for secure/non-secure targets.

ccli8 committed:
Convert TZ target name 'NPSA' to test spec platform name

1.  All TZ targets should have name pattern: PLATFORM_[PSA_/NPSA_]S/NS, where:
    (1) 'PLATFORM' for test spec platform name
    (2) 'PSA/NPSA' for PSA/non-PSA targets. Defaults to PSA target on absent.
    (3) 'S'/'NS' for secure/non-secure targets
2. Secure target may participate in Greentea, so its name is also truncated here.

urutva commented on tools/test_api.py:
```diff
@@ -2274,20 +2274,21 @@ def test_spec_from_test_builds(test_builds):
         # Convert TZ target name to test spec platform name
         #
-        # 1. All TZ targets should have name pattern: PLATFORM_[PSA_]S/NS, where:
+        # 1. All TZ targets should have name pattern: PLATFORM_[PSA_/NPSA_]S/NS, where:
```
@ccli8  Sorry for nitpicking. Having `PSA` in the target name is not mandatory as it is the default. I'd write this as
1. All TZ targets should have name pattern: PLATFORM_[NPSA_]S/NS, where:
(2) 'NPSA' for non-PSA targets. Defaults to PSA target if absent.
(3) 'S'/'NS' for secure/non-secure targets

0xc0170 replied:
@ccli8 Once updated, leave a comment, we restart CI

urutva commented on tools/test_api.py:
```diff
@@ -2274,20 +2274,21 @@ def test_spec_from_test_builds(test_builds):
                 test_builds[build]['platform'] = test_builds[build]['platform'][:-2]
-            
+
             if test_builds[build]['platform'].endswith('_PSA'):
```
@jamesbeyond  Since we do not have any PSA targets that ends with `_PSA` can we remove this check?

ccli8 replied:
Remove **PSA** check or keep it (no change)?

0xc0170 replied:
Possibly to remote, otherwise not blocking this PR. @ccli8 rather to check the above comment

ccli8 replied:
@0xc0170 @devran01 @jamesbeyond Removed **PSA** check from test spec name conversion

ccli8 committed:
Remove 'PSA' check from test spec conversion

This is unnecessary because all TZ targets must follow the naming rule: PLATFORM_[NPSA_]S/NS
